File: VB/WindowsApplication3/MyPropertyDescriptor.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.Windows.Forms
Imports DevExpress.Skins
Imports System.ComponentModel

Namespace DXSample
	Public Class MyPropertyDescriptor
		Inherits PropertyDescriptor
		Private pd As PropertyDescriptor

		Public Sub New(ByVal pd As PropertyDescriptor)
			MyBase.New(pd)
			Me.pd = pd
		End Sub

		Public Overrides ReadOnly Property Description() As String
			Get
				Return String.Format("=====Custom Description:======" & Constants.vbCrLf & " {0}", MyBase.Description)
			End Get
		End Property

		Public Overrides ReadOnly Property ComponentType() As Type
			Get
				Return pd.ComponentType
			End Get
		End Property
		Public Overrides ReadOnly Property IsReadOnly() As Boolean
			Get
				Return pd.IsReadOnly
			End Get
		End Property

		Public Overrides ReadOnly Property PropertyType() As Type
			Get
				Return pd.PropertyType
			End Get
		End Property

		Public Overrides Function CanResetValue(ByVal component As Object) As Boolean
			Return pd.CanResetValue(component)
		End Function

		Public Overrides Function GetValue(ByVal component As Object) As Object
			Return pd.GetValue(component)
		End Function
		Public Overrides Sub ResetValue(ByVal component As Object)
			pd.ResetValue(component)
		End Sub
		Public Overrides Sub SetValue(ByVal component As Object, ByVal value As Object)
			pd.SetValue(component, value)
		End Sub
		Public Overrides Function ShouldSerializeValue(ByVal component As Object) As Boolean
			Return pd.ShouldSerializeValue(component)
		End Function
	End Class
End Namespace

File: CS/WindowsApplication3/MyPropertyDescriptor.cs
using System;
using System.Collections.Generic;
using System.Windows.Forms;
using DevExpress.Skins;
using System.ComponentModel;

namespace DXSample {
    public class MyPropertyDescriptor : PropertyDescriptor
    {
        PropertyDescriptor pd;

        public MyPropertyDescriptor(PropertyDescriptor pd)
            : base(pd)
        {
            this.pd = pd;
        }

        public override string Description
        {
            get
            {
                return string.Format("=====Custom Description:======\r\n {0}", base.Description);
            }
        }

        public override Type ComponentType
        {
            get { return pd.ComponentType; }
        }
        public override bool IsReadOnly
        {
            get { return pd.IsReadOnly; }
        }

        public override Type PropertyType
        {
            get { return pd.PropertyType; }
        }

        public override bool CanResetValue(object component)
        {
            return pd.CanResetValue(component);
        }

        public override object GetValue(object component)
        {
            return pd.GetValue(component);
        }
        public override void ResetValue(object component)
        {
            pd.ResetValue(component);
        }
        public override void SetValue(object component, object value)
        {
            pd.SetValue(component, value);
        }
        public override bool ShouldSerializeValue(object component)
        {
            return pd.ShouldSerializeValue(component);
        }
    }
}